### PR TITLE
Added autograd tests, disabled jit autograd tests for complex and added a separate list for tests for complex dtype only

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38400 Add tan_cuda for complex dtypes
* **#38399 Added autograd tests, disabled jit autograd tests for complex and added a separate list for tests for complex dtype only**

Differential Revision: [D21555941](https://our.internmc.facebook.com/intern/diff/D21555941)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -238,6 +238,12 @@ def method_tests():
         ('log1p', uniform_scalar(requires_grad=True), NO_ARGS, 'scalar', (True,)),
         ('log2', torch.rand(S, S, S) + 1e-2, NO_ARGS, '', (True,)),
         ('log2', uniform_scalar(1e-2, requires_grad=True), NO_ARGS, 'scalar', (True,)),
+        ('log', torch.randn(S, S, S, dtype=torch.cfloat) + 1e-2, NO_ARGS, 'complex', (True,)),
```
Comment (albanD) [inline torch/testing/_internal/common_methods_invocations.py:241]: You have special cases here when the variant name contains "complex" right? Maybe add a comment at the beginning of this list mentioning it. And what is happening for tests that have "complex" in their variant name.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -273,6 +279,8 @@ def method_tests():
         ('atan2', (S, 1, S), ((S, S),), 'broadcast_all'),
         ('reciprocal', torch.rand(S, S, S) + 0.1, NO_ARGS, '', (True,)),
         ('reciprocal', uniform_scalar(0.1, requires_grad=True), NO_ARGS, 'scalar', (True,)),
+        ('reciprocal', torch.randn(S, S, S, dtype=torch.cdouble) + 0.1, NO_ARGS, 'complex', (True,)),
+        ('reciprocal', uniform_scalar(0.1j, requires_grad=True), NO_ARGS, 'complex_scalar', (True,)),
```
Comment (albanD) [inline torch/testing/_internal/common_methods_invocations.py:283]: I'm not sure why the original test set requires_grad=True here.
You don't do it for any other tests so I think you can remove it.

File: test/test_autograd.py [hunk 3]
```diff
@@ -4239,8 +4254,6 @@ def check(name):
                         output_variable = getattr(self_variable, name)(*args_variable, **kwargs_variable)
                         output_tensor = getattr(self_tensor, name)(*args_tensor, **kwargs_variable)
                         if not isinstance(output_tensor, torch.Tensor) and not istuple(output_tensor):
-                            # TODO: I'm not sure why we insert an outer dimension
```
Comment (albanD) [inline test/test_autograd.py:4242]: Just out of curiosity, what was the answer?

File: test/test_autograd.py [hunk 4]
```diff
@@ -4185,11 +4185,14 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
         test_case.assertEqualTypeString(self_variable, self_variable.grad)
         test_case.assertEqual(self_variable.size(), self_variable.grad.size())
 
+separate_complex_tests = ['log', 'log10', 'log1p', 'log2', 'reciprocal']
```
Comment (albanD) [inline test/test_autograd.py:4188]: Could you add a comment for what this list contains?
These are the ones that run only for complex?

File: test/test_autograd.py (hunk 3 above)
Comment (anjali411) [reply test/test_autograd.py:4242]: that case is executed when output_tensor is not a tensor or tuple, but a scalar and so that line just makes it a tensor

**Review by albanD — APPROVED** [review]
lgtm
